Tests/SwiftParserTest/StringLiteralContentValueTests.swift

```diff
@@ -0,0 +1,318 @@
+    test(#"\#u{1F600}"#)
+  }
+
+  // MARK: multi line string literal
+
+  func testMultiLine_emptyString() {
+    test(
+      """
+
+      """
+    )
+  }
+
+  func testMultiLine_emptyLine() {
+    test(
+      """
+
+
+      """
+    )
+  }
+
+  func testMultiLine_helloWorld() {
+    test(
+      """
+      Hello, world!
+      """
+    )
+  }
+
+  func testMultiLine_indentedLines() {
+    test(
+      """
+      not indented
+        2 spaces indented
+          4 spaces indented
+      """
+    )
+  }
+
+  func testMultiLine_escapedLine() {
+    test(
+      """
+      Line 1\
+      .still on Line 1
+      """
+    )
+  }
+
+  // MARK: raw multi line string literal
+
+  func testRawMultiLine_emptyString() {
+    test(
+      #"""
+
+      """#
+    )
+  }
+
+  func testRawMultiLine_emptyLine() {
+    test(
+      #"""
+
+
+      """#
+    )
+  }
+
+  func testRawMultiLine_helloWorld() {
+    test(
+      #"""
+      Hello, world!
+      """#
+    )
+  }
+
+  func testRawMultiLine_indentedLines() {
+    test(
+      #"""
+      not indented
+        2 spaces indented
+          4 spaces indented
+      """#
+    )
+  }
+
+  func testRawMultiLine_escapedLine() {
+    test(
+      #"""
+      Line 1\
+      .still on Line 1
+      """#
+    )
+  }
+
+  // MARK: literal value not available
+
+  func testMissingQuoteStringLiteral() throws {
+    let sut = StringLiteralExprSyntax(#""a"# as ExprSyntax)!
+    XCTAssertNil(sut.contentValue, "only fully parsed string literals should produce a literal value")
+  }
+
+  func testInterpolatedStringLiteral() throws {
+    let sut = StringLiteralExprSyntax(#""abc\(1)""# as ExprSyntax)!
+    XCTAssertNil(sut.contentValue, "interpolated string literals cannot produce a literal value")
+  }
+
+  func testMalformedMultiLineStringLiteral() throws {
+    let sut = StringLiteralExprSyntax(#""""a""""# as ExprSyntax)!
+    XCTAssertNil(sut.contentValue, "missing newline in multiline string literal cannot produce a literal value")
+  }
+
+  // MARK: supporting code
+
+  /// This helper function takes a string literal argument and compares its
+  /// value with the parsed node of type StringLiteralExprSyntax. To produce the
+  /// node it parses the contents of this source file and matches the literal by
+  /// position in source.
+  func test(_ expected: StaticString, file: StaticString = #filePath, line: UInt = #line) {
+    guard let literal = Self.literals[at: line] else {
+      fatalError("string literal not found at line \(line)")
+    }
+
+    guard let contentValue = literal.contentValue else {
+      XCTFail("literal unexpectedly product nil value", file: file, line: line)
+      return
+    }
+
+    XCTAssertEqual(contentValue, expected.description, file: file, line: line)
+  }
+
+  static let literals = try! StringLiteralCollector()
+
+  /// Helper class to find string literals in this source file
+  class StringLiteralCollector: SyntaxVisitor {
+    var locationConverter: SourceLocationConverter? = nil
```

Comment:
Does this need to be optional?

Reply:
It's only used temporarily during `walk` and can be freed afterwards. Not sure if this is a good reason.